scipy/stats/_fit.py

```diff
@@ -1254,6 +1270,27 @@ def _filliben(dist, data):
 _filliben.alternative = 'less'  # type: ignore[attr-defined]
 
 
+def _order_statistic_evs(dist, n):
+    # Calculate the expected values of the order statistics of a distribution
+    # PDF from https://en.wikipedia.org/wiki/Order_statistic
+    r = np.arange(1, n + 1)
+    constant = gammaln(n + 1) - gammaln(r) - gammaln(n - r + 1)
+    def integrand(x):
+        variable = dist.logpdf(x) + (r-1)*dist.logcdf(x) + (n-r)*dist.logsf(x)
+        return x * np.exp(constant + variable)
+
+    res = integrate.quad_vec(integrand, -np.inf, np.inf)
+    return res[0]
+
+
+def _shapiro_francia(dist, data):
+    n = data.shape[-1]
+    M = _order_statistic_evs(dist, n)
+    X = np.sort(data, axis=-1)
```

Comment:
it would seem like the `_order_statistic_evs` call would preserve order (or maybe reverse) but I didn't check closely, am I mistaken?

Reply:
`_order_statistic_evs` returns the expected values in the desired order. The `sort` is for the data itself.